validator/src/main.rs

```diff
@@ -282,6 +300,8 @@ fn get_rpc_node(
             let contact_info =
                 &eligible_rpc_peers[thread_rng().gen_range(0, eligible_rpc_peers.len())];
             return (contact_info.clone(), highest_snapshot_hash);
+        } else {
+            retry_reason = Some("No snapshots available".to_owned());
```

Comment:
I think this is more clearer place to put this message in terms of control flow..